src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,6 @@
+package model;
+
+import operation.OperationHandler;
+
+public record FruitTransaction(OperationHandler operation, String fruit, int amount) {
```

Comment:
A model should have a state, not a behavior
```suggestion
public record FruitTransaction(Operation operation, String fruit, int amount) {
```


```
    public enum Operation {
        BALANCE("b"),
        SUPPLY("s"),
        PURCHASE("p"),
        RETURN("r");

        private final String operation;

        Operation(String operation) {
            this.operation = operation;
        }

        public String getOperation() {
            return operation;
        }

        public static Operation getByCode(String character) {
            for (Operation operation : values()) {
                if (operation.getOperation().equals(character)) {
                    return operation;
                }
            }
            throw new RuntimeException("Unknown operation");
        }
    }
```